src/inspect_ai/tool/_tool_call.py

```diff
@@ -44,8 +44,11 @@ class ToolCall:
     arguments: dict[str, Any]
     """Arguments to function."""
 
-    type: Literal["function"]
-    """Type of tool call (currently only 'function')"""
+    type: str
+    """Type of tool call ('function' or a model specific built-in tool type)"""
+
+    native_name: str | None = field(default=None)
+    """Model's native name of the tool - if any."""
```

Comment:
This is the fundamental enhancement. I add an optional native model/native name into `ToolCall`. This will let us make the back transformation from inspect -> model whenever we want.